From 0d04a69d4f5bc02947eb9a17027b86147b556557 Mon Sep 17 00:00:00 2001
From: Alex Nedel <alex.nedel@opin.ca>
Date: Fri, 11 Jan 2013 11:38:42 -0500
Subject: [PATCH] Fixed replies to comments

---
 comment_goodness.module | 24 +++++++++++++++---------
 1 file changed, 15 insertions(+), 9 deletions(-)

diff --git a/comment_goodness.module b/comment_goodness.module
index cf339d7..e948e9c 100644
--- a/comment_goodness.module
+++ b/comment_goodness.module
@@ -130,7 +130,7 @@ function comment_goodness_node_type_form_submit($form, &$form_state) {
  * 1.1.1
  *
  *
- * REPLY ordering similar to Facbeook.
+ * REPLY ordering similar to Facebook.
  * 2
  * 2.0
  * 2.1
@@ -143,6 +143,10 @@ function comment_goodness_query_comment_filter_alter(QueryAlterableInterface $qu
   if (($node = $query->getMetaData('node')) && (get_class($query) == 'PagerDefault')) {
     // Get the configured default sort ordering for this node type.
     $sort = variable_get('comment_default_sorting_' . $node->type, comment_goodness_OLDER_FIRST);
+
+	// Get the configured default reply sort ordering for this node type.
+	$reply_sort = variable_get('comment_reply_default_sorting_' . $node->type, comment_goodness_OLDER_FIRST);
+
     // The default ordering is ASC (older on top). If the configured sorting is
     // newer on top, change the ordering to DESC.
     if ($sort == comment_goodness_NEWER_FIRST) {
@@ -153,13 +157,15 @@ function comment_goodness_query_comment_filter_alter(QueryAlterableInterface $qu
         // Get rid of the expressions that prepare the threads for ASC ordering.
         unset($expressions['torder']);
         unset($orderby['torder']);
-		if ($reply_sort == comment_goodness_NEWER_FIRST) {
-          // My addition for the reply commenting order stuff.
-	      $query->addExpression('SUBSTRING(c.thread, 1, 2)', 'corder');
-	      $query->orderBy('corder', 'DESC');
-	      $query->addExpression('SUBSTRING(c.thread, 1, (LENGTH(c.thread) - 1))', 'rorder');
-	      $query->orderBy('rorder', 'ASC'); 
-        }
+		
+		// Sorting for replies to comments
+		if ($reply_sort == comment_goodness_OLDER_FIRST) {
+			      $query->addExpression('SUBSTRING(c.thread, 1, 2)', 'corder');
+			      $query->orderBy('corder', 'DESC');
+			      $query->addExpression('SUBSTRING(c.thread, 1, (LENGTH(c.thread) - 1))', 'rorder');
+			      $query->orderBy('rorder', 'ASC'); 
+		        }
+		
 		// Simply order by the thread field.
 		else {
           $orderby['c.thread'] = 'DESC';
@@ -793,4 +799,4 @@ function _comment_goodness_get_submitted_string($date_type) {
     $submitted_string = 'Submitted by !username on !datetime';
   }
   return $submitted_string;
-}
+}
\ No newline at end of file
-- 
1.7.11.1

