diff --git a/modules/salesforce_mapping/salesforce_mapping.module b/modules/salesforce_mapping/salesforce_mapping.module
index 9c0c8f6..85089b4 100644
--- a/modules/salesforce_mapping/salesforce_mapping.module
+++ b/modules/salesforce_mapping/salesforce_mapping.module
@@ -171,6 +171,13 @@ function salesforce_mapping_menu() {
     $info = entity_get_info($type);
     $bundle_key = $info['entity keys']['bundle'] ?: 'type';
     $entity = entity_create($type, array($bundle_key => $bundle));
+
+    $info = entity_get_info($type);
+    $entity_id_field = $info['entity keys']['id'];
+    if (!isset($entity->$entity_id_field)) {
+      $entity->$entity_id_field = '';
+    }
+
     $uri = method_exists($entity, 'uri') ? $entity->uri() : entity_uri($type, $entity);
     if (empty($uri['path'])) {
       $path = 'admin/content/salesforce/' . $type . '/%' . $type . '/salesforce_activity';
