diff --git a/ctools.module b/ctools.module
index 78ac26b..6d55396 100644
--- a/ctools.module
+++ b/ctools.module
@@ -758,12 +758,6 @@ function ctools_process(&$variables, $hook) {
     $variables['classes_array'] = array_diff($variables['classes_array'], $remove_classes);
   }
 
-  // Update the classes within the attributes array to match the classes array
-  if (isset($variables['attributes_array']['class'])) {
-    $variables['attributes_array']['class'] = array_unique(array_merge($variables['classes_array'], $variables['attributes_array']['class']));
-    $variables['attributes'] = $variables['attributes_array'] ? drupal_attributes($variables['attributes_array']) : '';
-  }
-
   // Since this runs after template_process(), we need to re-implode the
   // classes array.
   $variables['classes'] = implode(' ', $variables['classes_array']);
